cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1160,6 +1160,13 @@ private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> onReady(Reconcilia
                                 new NoSuchResourceException("Kafka resource '" + clusterName
                                         + "' identified by label '" + Labels.STRIMZI_CLUSTER_LABEL
                                         + "' does not exist in namespace " + clusterNamespace + ".")).mapEmpty();
+                    } else if (kafka.getStatus().getConditions().stream().noneMatch(condition -> condition.getType().equals("Ready") && condition.getStatus().equals("True"))) {
```

Comment:
Looks like this comment got lost somewhere while I was doing the review ... I think this smells after NPE. What if the status or the conditions are null? That can easily happen in situations like this. (and it would indicate that it is not ready as well)

Reply:
Okay, I will include 2 more condition in the `else if`. Something like this --> ```        else if (kafka.getStatus() == null 
                || kafka.getStatus().getConditions() 
                || kafka.getStatus().getConditions().stream().noneMatch(condition -> condition.getType().equals("Ready") && condition.getStatus().equals("True"))){```